From 31322c0b759711fd7eb5348f8b05663a93ca5785 Mon Sep 17 00:00:00 2001
From: Giulio Fidente <gfidente@redhat.com>
Date: Sun, 14 Jun 2015 10:18:41 +0200
Subject: [PATCH] Allow customization of dhcp_domain setting from
 agents/dhcp.pp

With this patch we aim at making the dhcp_domain setting
customizable from the agents/dhcp class.

Change-Id: I457ae90cb8de500998e9d4775f2832747d052628
---
 neutron/manifests/agents/dhcp.pp                 | 6 ++++++
 neutron/spec/classes/neutron_agents_dhcp_spec.rb | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/neutron/manifests/agents/dhcp.pp b/neutron/manifests/agents/dhcp.pp
index 5025e04..ab63635 100644
--- a/neutron/manifests/agents/dhcp.pp
+++ b/neutron/manifests/agents/dhcp.pp
@@ -29,6 +29,10 @@
 # [*interface_driver*]
 #   (optional) Defaults to 'neutron.agent.linux.interface.OVSInterfaceDriver'.
 #
+# [*dhcp_domain*]
+#   (optional) domain to use for building the hostnames
+#   Defaults to 'openstacklocal'
+#
 # [*dhcp_driver*]
 #   (optional) Defaults to 'neutron.agent.linux.dhcp.Dnsmasq'.
 #
@@ -69,6 +73,7 @@ class neutron::agents::dhcp (
   $state_path             = '/var/lib/neutron',
   $resync_interval        = 30,
   $interface_driver       = 'neutron.agent.linux.interface.OVSInterfaceDriver',
+  $dhcp_domain            = 'openstacklocal',
   $dhcp_driver            = 'neutron.agent.linux.dhcp.Dnsmasq',
   $root_helper            = 'sudo neutron-rootwrap /etc/neutron/rootwrap.conf',
   $use_namespaces         = true,
@@ -113,6 +118,7 @@ class neutron::agents::dhcp (
     'DEFAULT/state_path':             value => $state_path;
     'DEFAULT/resync_interval':        value => $resync_interval;
     'DEFAULT/interface_driver':       value => $interface_driver;
+    'DEFAULT/dhcp_domain':            value => $dhcp_domain;
     'DEFAULT/dhcp_driver':            value => $dhcp_driver;
     'DEFAULT/use_namespaces':         value => $use_namespaces;
     'DEFAULT/root_helper':            value => $root_helper;
diff --git a/neutron/spec/classes/neutron_agents_dhcp_spec.rb b/neutron/spec/classes/neutron_agents_dhcp_spec.rb
index ff6b973..8c880d6 100644
--- a/neutron/spec/classes/neutron_agents_dhcp_spec.rb
+++ b/neutron/spec/classes/neutron_agents_dhcp_spec.rb
@@ -17,6 +17,7 @@ describe 'neutron::agents::dhcp' do
       :state_path             => '/var/lib/neutron',
       :resync_interval        => 30,
       :interface_driver       => 'neutron.agent.linux.interface.OVSInterfaceDriver',
+      :dhcp_domain            => 'openstacklocal',
       :dhcp_driver            => 'neutron.agent.linux.dhcp.Dnsmasq',
       :root_helper            => 'sudo neutron-rootwrap /etc/neutron/rootwrap.conf',
       :use_namespaces         => true,
@@ -41,6 +42,7 @@ describe 'neutron::agents::dhcp' do
       is_expected.to contain_neutron_dhcp_agent_config('DEFAULT/state_path').with_value(p[:state_path]);
       is_expected.to contain_neutron_dhcp_agent_config('DEFAULT/resync_interval').with_value(p[:resync_interval]);
       is_expected.to contain_neutron_dhcp_agent_config('DEFAULT/interface_driver').with_value(p[:interface_driver]);
+      is_expected.to contain_neutron_dhcp_agent_config('DEFAULT/dhcp_domain').with_value(p[:dhcp_domain]);
       is_expected.to contain_neutron_dhcp_agent_config('DEFAULT/dhcp_driver').with_value(p[:dhcp_driver]);
       is_expected.to contain_neutron_dhcp_agent_config('DEFAULT/root_helper').with_value(p[:root_helper]);
       is_expected.to contain_neutron_dhcp_agent_config('DEFAULT/use_namespaces').with_value(p[:use_namespaces]);
